### PR TITLE
Calomel now actually does as described

### PR DESCRIPTION
**Pull request by Time-Green**
:cl:
fix: calomel now actually purges all chems, instead of only toxins
/:cl:

Not sure what to label this as, since I'm not sure if the description is wrong or the functionality is wrong. 

The wiki and the chems description all describe calomel removing ALL reagents, which could be very useful when dealing with any harmful amount of chems, but it only removed toxin reagents

### CODE REVIEW

File: code/modules/reagents/chemistry/reagents/medicine_reagents.dm [hunk 1]
```diff
@@ -430,8 +430,9 @@
 	chemical_flags = REAGENT_CAN_BE_SYNTHESIZED
 
 /datum/reagent/medicine/calomel/on_mob_life(mob/living/carbon/M, delta_time, times_fired)
-	for(var/datum/reagent/toxin/R in M.reagents.reagent_list)
-		M.reagents.remove_reagent(R.type, 3 * REM * delta_time)
+	for(var/datum/reagent/R in M.reagents.reagent_list)
+		if(istype(R.type, /datum/reagent/medicine/calomel))
```
Comment (TemporalOroboros) [inline code/modules/reagents/chemistry/reagents/medicine_reagents.dm:434]:
```suggestion
		if(!istype(R.type, /datum/reagent/medicine/calomel))
```

Comment (TemporalOroboros) [reply code/modules/reagents/chemistry/reagents/medicine_reagents.dm:434]: Currently this just purges itself.

Comment (Time-Green) [reply code/modules/reagents/chemistry/reagents/medicine_reagents.dm:434]: im convinced vscode is intentionally inverting my code

Comment (TemporalOroboros) [reply code/modules/reagents/chemistry/reagents/medicine_reagents.dm]: Also, you are typechecking the iterator reagent's type, not the iterator reagent itself.

Comment (Time-Green) [reply code/modules/reagents/chemistry/reagents/medicine_reagents.dm]: This is what happens to a mf when he changes if(R.type != /datum/reagent/medicine/calomel) at the last second

**Comment by Krysonism** [comment]
I think this change was an intentional part of the PR that massively power creeped all the toxin meds.

Calomel won't be very useful either way though.

**Comment by Time-Green** [comment]
Right now it’s just an inferior toxin purging chem. With this change it could also fix medical overdoses, cancel other peoples healing meds and fix the funny thing where people will drink 1000u of beer. That while still dealing toxin damage

It’s a niche chem either way, but this way it has some more situational value

I have no idea what the intent was, I’m content to see what happens

**Comment by Krysonism** [comment]
> 
> 
> Right now it’s just an inferior toxin purging chem. With this change it could also fix medical overdoses, cancel other peoples healing meds and fix the funny thing where people will drink 1000u of beer. That while still dealing toxin damage
> 
> It’s a niche chem either way, but this way it has some more situational value
> 
> I have no idea what the intent was, I’m content to see what happens

multiver can also fix medicine and beer overdoses. At the same speed and healing instead of causing toxin damage.

**Comment by Time-Green** [comment]
But what if you have multiple medical overdoses? Who you gonna call then? (because multiver only purges if there's less than two or three unique meds)

**Comment by Krysonism** [comment]
> But what if you have multiple medical overdoses? Who you gonna call then? (because multiver only purges if there's less than two or three unique meds)

anacea?

I'd love for calomel to be useful again., I have fond memories of using it before multiver was buffed. Not sure if this is the way to go though.

**Comment by Time-Green** [comment]
I like the idea of it being a toxic, relentless, cheap but reliable chem purger. You could whip up tons of the stuff to combat spider invasions, overdoses and everything

Some other stuff could be fun, but I can’t think of anything rn

**Comment by Time-Green** [comment]
I changed my mind

**Pull request closed without merging**